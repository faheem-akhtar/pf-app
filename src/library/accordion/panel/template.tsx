import { FunctionComponent } from 'react';

import styles from '../accordion.module.scss';
import { AccordionPanelTemplatePropsInterface } from './template-props.interface';

export const AccordionPanelTemplate: FunctionComponent<AccordionPanelTemplatePropsInterface> = ({
  listItem = false,
  children,
}) => {
  const Component = listItem ? 'li' : 'div';

  return (
    <Component className={styles.accordion__panel} role={listItem ? 'listitem' : undefined}>
      {children}
    </Component>
  );
};
