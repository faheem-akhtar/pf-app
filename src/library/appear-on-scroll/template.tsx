import { domClassMerge } from 'helpers/dom/class-merge';

import { AppearOnScrollStatusEnum } from './status.enum';
import { AppearOnScrollTemplatePropsInterface } from './template-props.interface';

import styles from './appear-on-scroll.module.scss';

const classByStatus = {
  [AppearOnScrollStatusEnum.ENTERING]: styles.appear_on_scroll__entering,
  [AppearOnScrollStatusEnum.EXITING]: styles.appear_on_scroll__exiting,
  [AppearOnScrollStatusEnum.WRAPPER_IN_VIEW]: styles.appear_on_scroll__hidden,
  [AppearOnScrollStatusEnum.WRAPPER_OUTSIDE_VIEW]: styles.appear_on_scroll__visible,
};

export const AppearOnScrollTemplate = ({
  status,
  className,
  children,
}: AppearOnScrollTemplatePropsInterface): JSX.Element => (
  <div
    data-testid='AppearOnScrollTemplate'
    className={domClassMerge(styles.appear_on_scroll, classByStatus[status as AppearOnScrollStatusEnum], className)}
  >
    {children}
  </div>
);
