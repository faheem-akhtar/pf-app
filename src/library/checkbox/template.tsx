import { FunctionComponent } from 'react';

import { domClassMerge } from 'helpers/dom/class-merge';

import styles from './checkbox.module.scss';
import { CheckboxTemplatePropsInterface } from './template-props.interface';

/**
 * Checkbox allows users to select one or more options from the given options.
 *
 * Design: https://www.figma.com/file/gRARY1Vi4W2Ow1vRLw02am/PF_Consumer-Web-Kit?node-id=6%3A1581
 */
export const CheckboxTemplate: FunctionComponent<CheckboxTemplatePropsInterface> = (props): JSX.Element => (
  <div className={domClassMerge(styles.container, props.containerClassName)}>
    <input
      id={props.id}
      name={props.name}
      aria-labelledby={props.name}
      type='checkbox'
      checked={props.checked}
      onChange={props.onChange}
      className={domClassMerge(styles.input, props.inputClassName)}
    />
    <label htmlFor={props.id} className={domClassMerge(styles.label, props.labelClassName)}>
      {props.children}
    </label>
  </div>
);
