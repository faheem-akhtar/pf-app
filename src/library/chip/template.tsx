import { domClassMerge } from 'helpers/dom/class-merge';

import styles from './chip.module.scss';
import { ChipTemplatePropsInterface } from './template-props.interface';

export const ChipTemplate = (props: ChipTemplatePropsInterface): JSX.Element => {
  return (
    <label
      htmlFor={props.htmlFor}
      className={domClassMerge(styles.chip, props.className, {
        [styles.chip_disabled]: props.isDisabled,
        [styles.chip_selected]: props.isSelected,
      })}
      onClick={props.onClick}
    >
      {props.prefixIcon && <span className={styles.chipPrefix}>{props.prefixIcon}</span>}
      {<span className={domClassMerge(styles.chipLabel, props.labelClassName)}>{props.label}</span>}
      {props.suffixIcon && <span className={styles.chipSuffix}>{props.suffixIcon}</span>}
    </label>
  );
};
