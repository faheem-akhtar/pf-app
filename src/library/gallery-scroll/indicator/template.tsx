import { domClassMerge } from 'helpers/dom/class-merge';

import { galleryScrollIndicatorGetTransform } from './get-transform';
import { galleryScrollIndicatorMakeItems } from './make-items';
import { galleryScrollIndicatorVisibleItemsCount } from './visible-items-count';

import { GalleryScrollIndicatorItemInterface } from './item.interface';
import { GalleryScrollIndicatorPropsInterface } from './props.interface';

import styles from './gallery-scroll-indicator.module.scss';

export const GalleryScrollIndicatorTemplate = (props: GalleryScrollIndicatorPropsInterface): JSX.Element | null => {
  if (props.itemsCount <= 1) return null;

  const items = galleryScrollIndicatorMakeItems(props.itemsCount, props.activeIndex);

  const getItemClassName = (item: GalleryScrollIndicatorItemInterface): string =>
    domClassMerge(styles.dot, {
      [styles.dot__active]: item.isActive,
      [styles.dot__hidden]: item.isHidden,
    });

  const transformXPx = galleryScrollIndicatorGetTransform({ items, itemWidth: props.itemWidth });
  const containerStyle = {
    transform: `translateX(${transformXPx}px)`,
  };

  return (
    <div className={styles.container} style={{ width: props.itemWidth * galleryScrollIndicatorVisibleItemsCount }}>
      <div className={styles.items_container} style={containerStyle}>
        {items.map((item, itemIndex) => (
          <div key={itemIndex} className={getItemClassName(item)} />
        ))}
      </div>
    </div>
  );
};
