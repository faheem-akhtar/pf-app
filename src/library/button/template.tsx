import { Fragment, FunctionComponent } from 'react';

import { domClassMerge } from 'helpers/dom/class-merge';

import { ButtonIconPositionEnum } from './icon-position.enum';
import { ButtonLinkTemplatePropsInterface } from './link-template-props.interface';
import { ButtonTemplatePropsInterface } from './template-props.interface';
import { LoaderTemplate } from 'components/loader/template';

import styles from './button.module.scss';

// TODO-FE[TPNX-3016] add tests
/**
 * Web UI Kit Button tempalte
 * Figma: https://www.figma.com/file/gRARY1Vi4W2Ow1vRLw02am/PF_Consumer-Web-Kit?node-id=3869%3A2340
 */
export const ButtonTemplate: FunctionComponent<ButtonLinkTemplatePropsInterface | ButtonTemplatePropsInterface> = ({
  icon = {},
  componentType,
  ...props
}) => {
  const { component: Icon, position: iconPosition } = icon;

  const renderLoader = (): JSX.Element => (
    <LoaderTemplate containerCssClass={styles.loader} dotCssClass={styles[`${componentType}--loader`]} />
  );

  const renderIcon = (): JSX.Element | null =>
    Icon ? (
      <Icon clipped={false} class={domClassMerge(styles.icon, styles[`icon--${iconPosition}`], icon.className)} />
    ) : null;

  const renderContent = (): JSX.Element => (
    <Fragment>
      {(iconPosition === ButtonIconPositionEnum.left || iconPosition === ButtonIconPositionEnum.top) && renderIcon()}
      {props.children}
      {iconPosition === ButtonIconPositionEnum.right && renderIcon()}
    </Fragment>
  );

  const Component = (props as ButtonLinkTemplatePropsInterface).href ? 'a' : 'button';

  return (
    <Component
      class={domClassMerge(
        styles.button,
        styles[componentType],
        styles[props.size],
        {
          [styles[`${componentType}--loading`]]: !!props.loading,
          [styles.disabled]: !!props.disabled,
          [styles.loading]: !!props.loading,
          [styles[`${componentType}--disabled`]]: !!props.disabled,
        },
        props.className
      )}
      disabled={props.disabled}
      onClick={!props.loading ? props.onClick : undefined}
      href={(props as ButtonLinkTemplatePropsInterface).href}
      target={(props as ButtonLinkTemplatePropsInterface).target}
    >
      {props.loading ? renderLoader() : renderContent()}
    </Component>
  );
};
