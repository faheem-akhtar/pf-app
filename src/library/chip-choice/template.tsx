import { useTranslationHook } from 'helpers/hook/translation.hook';

import { ChipChoiceTemplatePropsInterface } from './template-props.interface';
import { ChipTemplate } from '../chip/template';

import styles from './chip-choice.module.scss';

/**
 * Chip choice allows users to make selections from a list of values.
 *
 * Design: https://www.figma.com/file/gRARY1Vi4W2Ow1vRLw02am/PF_Consumer-Web-Kit?node-id=16%3A781
 */
export const ChipChoiceTemplate = <V extends unknown>(props: ChipChoiceTemplatePropsInterface<V>): JSX.Element => {
  const { t } = useTranslationHook();
  const { placeholder = t('any') as string } = props;

  return (
    <ul className={props.containerClassName || ''}>
      {props.options.map((option, id) => (
        <li key={id} className={styles.item}>
          <ChipTemplate
            label={option.label ? option.label : placeholder}
            className={props.chipClassName}
            isSelected={props.selected === option.value}
            onClick={(e): void => {
              props.onCheck(option, e);
            }}
          />
        </li>
      ))}
    </ul>
  );
};
