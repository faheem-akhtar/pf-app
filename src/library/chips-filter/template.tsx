import { ChipsFilterIconPositionEnum } from './icon-position.enum';
import { ChipsFilterTemplatePropsInterface } from './template-props.interface';
import { ChipTemplate } from 'library/chip/template';

import { IconThinCheckmarkTemplate } from 'components/icon/thin/checkmark-template';

import styles from './chips-filter.module.scss';

/**
 * Chip filter allow users to make multiple selections from a list of values.
 *
 * Design: https://www.figma.com/file/gRARY1Vi4W2Ow1vRLw02am/PF_Consumer-Web-Kit?node-id=16%3A781
 */
export const ChipsFilterTemplate = <V extends string>(props: ChipsFilterTemplatePropsInterface<V>): JSX.Element => {
  const {
    options,
    selectedIcon = IconThinCheckmarkTemplate,
    iconPosition = ChipsFilterIconPositionEnum.start,
    containerClassName,
  } = props;
  const Icon = selectedIcon;

  return (
    <ul className={containerClassName ? containerClassName : ''}>
      {options.map((option, id) => {
        const isSelected = !!props.selected.find((value) => value === option.value);

        return (
          <li className={styles.item} key={id}>
            <ChipTemplate
              label={option.label}
              prefixIcon={
                isSelected && iconPosition === ChipsFilterIconPositionEnum.start ? (
                  <Icon class={styles.icon} />
                ) : undefined
              }
              suffixIcon={
                isSelected && iconPosition === ChipsFilterIconPositionEnum.end ? (
                  <Icon class={styles.icon} />
                ) : undefined
              }
              isSelected={isSelected}
              onClick={(e): void => {
                // Check if element already selected
                let newValue: Array<V>;
                if (isSelected) {
                  newValue = props.selected.filter((value) => value !== option.value);
                } else {
                  newValue = [...props.selected, option.value];
                }

                props.onCheck(newValue, e);
              }}
            />
          </li>
        );
      })}
    </ul>
  );
};
