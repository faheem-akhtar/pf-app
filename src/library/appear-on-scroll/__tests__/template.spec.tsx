import { shallow } from 'enzyme';

import { AppearOnScrollStatusEnum } from '../status.enum';
import { AppearOnScrollTemplate } from '../template';

describe('AppearOnScrollTemplate', () => {
  it('should render template without an error', () => {
    const wrapper = shallow(
      <AppearOnScrollTemplate
        status={AppearOnScrollStatusEnum.ENTERING}
        className={'additional class'}
        children={'123'}
      />
    );

    expect(wrapper.html()).toMatchSnapshot();
  });
});
