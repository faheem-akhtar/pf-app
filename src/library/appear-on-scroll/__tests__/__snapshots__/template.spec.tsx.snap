// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppearOnScrollTemplate should render template without an error 1`] = `"<div class=\\"additional class\\">123</div>"`;
